### PR TITLE
Add Railway as Deployment Platform

### PR DESCRIPTION
**Pull request by sbmsr**
I was able to deploy to railway for free. Thought it would be useful to add the deploy to railway button.

Note: I have included my referral code in the url. I am perfectly fine with removing it. lmk!

### CODE REVIEW

**Comment by bewest** [comment]
@sbmsr Great idea, thanks!  You'll need to create a pull request on the `dev` branch, which is where we prep all the releases.  I have some concerns about the potential quality of service at railway on the free trial.  Does paying the $5 secure a reliable experience for Nightscout? https://github.com/nightscout/nightscout.github.io/issues/8

@sulkaharo We'll need to consider a bevvy of buttons and potentially additional template files.  We could plan on a `nightscout-in-cloud` repo that maintains all deployment templates and button definitions if we don't want them in the core repo?  There's interest in digital ocean, fly.io and others as well, not to mention variants with sidecars like tconnectsync.

File: README.md [hunk 1]
```diff
@@ -50,6 +52,8 @@ See [CONTRIBUTING.md](CONTRIBUTING.md)
...
+[railway-url]:https://railway.app/new/template?template=https%3A%2F%2Fgithub.com%2Fnightscout%2Fcgm-remote-monitor&plugins=mongodb&envs=API_SECRET%2CAUTH_DEFAULT_ROLES%2Cenable%2CMONGODB_URI%2CDISPLAY_UNITS&optionalEnvs=MONGODB_URI&API_SECRETDesc=A+passphrase+that+must+be+at+least+12+characters+long.&AUTH_DEFAULT_ROLESDesc=Possible+values+readable%2C+denied%2C+or+any+valid+role+name.+When+readable%2C+anyone+can+view+Nightscout+without+a+token.+Setting+it+to+denied+will+require+a+token+from+every+visit%2C+using+status-only+will+enable+api-secret+based+login.&enableDesc=Plugins+to+enable+for+your+site.+Must+be+a+space-delimited%2C+lower-case+list.&MONGODB_URIDesc=The+connection+string+required+to+access+your+database+%28where+all+your+data+is+stored+in+the+cloud%29.&DISPLAY_UNITSDesc=Preferred+BG+units+for+the+site%3A+mg%2Fdl+or+mmol%2FL+%28or+just+mmol%29.+Setting+to+mmol%2FL+puts+the+entire+server+into+mmol%2FL+mode+by+default%2C+no+further+settings+needed.&AUTH_DEFAULT_ROLESDefault=denied&enableDefault=careportal+basal+dbsize+rawbg+iob+maker+cob+bwp+cage+iage+sage+boluscalc+pushover+treatmentnotify+loop+pump+profile+food+openaps+bage+alexa+override+speech+cors&referralCode=9Y13IY
```
Comment (sulkaharo) [inline README.md:56]: referralCode=9Y13IY ?

**Comment by asonnera** [comment]
> I was able to deploy to railway for free. Thought it would be useful to add the deploy to railway button.
> 
> Note: I have included my referral code in the url. I am perfectly fine with removing it. lmk!

@sbmsr did you try to deploy in a newly created empty GitHub account like most new Nightscout users will try to? From what I've seen the deploy scripts interrupts as it needs verification and leaves you a 200 hours account. When relaunching with a cgm-remote-monitor fork it validates the account and gives 500 hours but still asks you a lot of stuff. Bottom line this button only works once you've setup your Railway account with Github. Can you try and confirm? Note I'm @psonnera and using my son's original account from the time I didn't know how things worked.

File: README.md (hunk 1 above)
Comment (sbmsr) [reply README.md]: Hi - yes, i added my referral code, as mentioned in the initial comment [here](https://github.com/nightscout/cgm-remote-monitor/pull/7319#issue-1122666169). I'm perfectly fine with removing it though :)

**Comment by sbmsr** [comment]
> Consider moving buttons like this to the documentation site. We've created an initial stub: [https://github.com/nightscout/nightscout.github.io/tree/source/docs/vendors/railway.app](https://github.com/nightscout/nightscout.github.io/tree/source/docs/vendors/railway.app?rgh-link-date=2022-09-01T15%3A32%3A54Z)

Hi - thanks for the heads up. So should I make a new pr to [nightscout/nightscout.github.io](https://github.com/nightscout/nightscout.github.io)? or should I modify this one to point to dev?

**Comment by sbmsr** [comment]
> > I was able to deploy to railway for free. Thought it would be useful to add the deploy to railway button.
> > Note: I have included my referral code in the url. I am perfectly fine with removing it. lmk!
> 
> @sbmsr did you try to deploy in a newly created empty GitHub account like most new Nightscout users will try to? From what I've seen the deploy scripts interrupts as it needs verification and leaves you a 200 hours account. When relaunching with a cgm-remote-monitor fork it validates the account and gives 500 hours but still asks you a lot of stuff. Bottom line this button only works once you've setup your Railway account with Github. Can you try and confirm? Note I'm @psonnera and using my son's original account from the time I didn't know how things worked.

It looks like they've updated their templates, so I'm going to have to generate a new link and test it out.  I'll try this out with a fresh account tomorrow and get back to you.

**Comment by bewest** [comment]
Yes, please use https://github.com/nightscout/nightscout.github.io for work on this going forward.

**Pull request closed without merging**